spi/topology/src/main/java/quarks/topology/spi/graph/ConnectorStream.java

```diff
@@ -163,6 +175,100 @@ protected Graph graph() {
     }
     
     @Override
+    public <J, U, K> TStream<J> join(Function<T, K> keyer,
+            TWindow<U, K> twindow, BiFunction<T, List<U>, J> joiner) {
+        
+        if(twindow instanceof TWindowImpl){
+            TStream<U> lastStream = twindow.feeder();
```

Comment:
There's several lines of code that are common between the two implementations, can they be shared?